### PR TITLE
Improve factus docs for Optimistic Locking and waitFor with SubscribedProjections

### PR DESCRIPTION
**Pull request by olewehrmeyer**
This PR adds documentation for

 - using the `resultFn` of an `attempt` function
 - using the `waitFor` feature for subscribed projections

It also
- updates projection `serial` to `revision` to stop advertising deprecated fields
- links the script used for local site development in the README

### CODE REVIEW

File: factcast-site/documentation-docsy/content/en/Usage/factus/optimistic-locking.md [hunk 1]
```diff
@@ -123,5 +123,31 @@ Also note that you should not (and cannot) publish to Factus directly when execu
 potentially
 break the purpose of the optimistic lock, and can lead to infinite loops.
 
-For further details on how to add operations that are executed after successful publishing or on failure handling,
+In certain cases you might want to access the facts that were published inside the attempt block. Similar to the
+`org.factcast.factus.Factus#publish` method that has overloads where you can specify a `Function<Fact, T> resultFn`,
+you can pass a similar `resultFn` or simply a `Runnable` to the `attempt` method. After successful publication this
+function will be called with a `List<Fact>` containing the published facts (in order they were published in). The return
```
Comment (bedaka) [inline factcast-site/documentation-docsy/content/en/Usage/factus/optimistic-locking.md:129]:
```suggestion
function will be called with a `List<Fact>` containing the published facts (in the order of publication). The return
```

File: factcast-site/documentation-docsy/content/en/Usage/factus/projections/types/subscribed-projection.md [hunk 2]
```diff
@@ -33,3 +33,29 @@ A `SubscribedProjection` is also `StateAware` and `WriterTokenAware`. However, t
...
+
+## Read-After-Write Consistency
+
+As a subscribed projection, factus will update the projections automatically in the background. A manual update with
+`factus.update(projection)` is not possible. In some cases however it might still be necessary to make sure a subscribed
+projection has processed a fact before continuing.
+
+One such use-case might be read-after-write consistency. Imagine a projection powering a table shown to a user. This 
+table shows information collected from facts `A` and `B`, where `B` gets published by the current application, but 
+`A` is published by another service. With the push of a button a user can publish a new `B` fact, creating another row
```
Comment (bedaka) [inline factcast-site/documentation-docsy/content/en/Usage/factus/projections/types/subscribed-projection.md:45]:
```suggestion
`A` is published by another service, which means we need to use a subscribed projection. With the push of a button a user can publish a new `B` fact, creating another row 
```

File: factcast-site/documentation-docsy/content/en/Usage/factus/projections/types/subscribed-projection.md [hunk 3]
```diff
@@ -33,3 +33,29 @@ A `SubscribedProjection` is also `StateAware` and `WriterTokenAware`. However, t
...
+projection has processed a fact before continuing.
+
+One such use-case might be read-after-write consistency. Imagine a projection powering a table shown to a user. This 
+table shows information collected from facts `A` and `B`, where `B` gets published by the current application, but 
+`A` is published by another service. With the push of a button a user can publish a new `B` fact, creating another row 
+in the table. If your frontend then immediately reloads the table, it might not yet show the new row, as the subscribed 
+projection has not yet processed the new fact.
+
+In this case you can use the `factus.waitFor` method to wait until the projection has consumed a certain fact. This 
+method will block until the fact is either procecced or the timeout is exceeded.
```
Comment (bedaka) [inline factcast-site/documentation-docsy/content/en/Usage/factus/projections/types/subscribed-projection.md:50]:
```suggestion
method will block until the fact is either processed or the timeout is exceeded.
```

File: factcast-site/documentation-docsy/content/en/Usage/factus/projections/types/subscribed-projection.md [hunk 4]
```diff
@@ -33,3 +33,29 @@ A `SubscribedProjection` is also `StateAware` and `WriterTokenAware`. However, t
...
+
+```java
+// publish a fact we need to wait on and extract its ID
+final var factId = factus.publish(new BFact(), Fact::id);
+
+factus.waitFor(subscribedProjection, factId, Duration.ofSeconds(5));
+```
+
+With this, the waiting thread will block for up to 5 seconds or until the projection has processed the fact stream until or beyond the specified fact.
```
Comment (bedaka) [inline factcast-site/documentation-docsy/content/en/Usage/factus/projections/types/subscribed-projection.md:59]:
```suggestion
With this, the waiting thread will block for up to 5 seconds or until the projection has processed the fact stream up to or beyond the specified fact.
```

**Comment by otbe** [comment]
Tracked in https://github.com/factcast/factcast/pull/3172

**Pull request closed without merging**